File: paobar_java/src/main/java/com/sosocom/entity/Song.java
package com.sosocom.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

/**
 * 歌曲实体类
 */
@Data
@TableName("song")
public class Song {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String songName;

    /** 歌名拼音排序键（全拼小写），列表分页按此字段排序。 */
    private String sortKey;

    private String artist;

    private String originalUrl;

    private String lyrics;

    /** 规范化吉他谱 JSON 字符串（对应表字段 tab_content_json）。 */
    private String tabContentJson;

    private String tabImageUrl;

    private String meta;

    private String difficulty;

    private String tuning;

    private Integer capo;

    private String playKey;

    private String originalKey;

    private String beat;

    private Integer viewCount;

    private Integer favoriteCount;

    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: paobar_fluter/lib/features/songs/domain/entities/song.dart
import 'package:equatable/equatable.dart';
import 'package:paobar/features/now_playing/presentation/widgets/tab_renderer/tab_document.dart';

/// 对齐后端 `SongDTO`。注意：原来的 HTML 字段 `tabContent` 已被移除，改成结构化的
/// [tabDocument]（爬虫入库时规范化生成），前端不再做 HTML 解析。
class Song extends Equatable {
  const Song({
    required this.id,
    required this.title,
    required this.artist,
    this.originalUrl,
    this.lyrics,
    this.tabDocument,
    this.tabImageUrl,
    this.meta,
    this.difficulty,
    this.tuning,
    this.capo,
    this.playKey,
    this.originalKey,
    this.beat,
    this.viewCount = 0,
    this.favoriteCount = 0,
  });

  final String id;
  final String title;
  final String artist;
  final String? originalUrl;
  final String? lyrics;
  final TabDocument? tabDocument;
  final String? tabImageUrl;
  final String? meta;
  final String? difficulty;
  final String? tuning;
  final int? capo;
  final String? playKey;
  final String? originalKey;
  final String? beat;
  final int viewCount;
  final int favoriteCount;

  @override
  List<Object?> get props => [
        id,
        title,
        artist,
        originalUrl,
        lyrics,
        tabDocument,
        tabImageUrl,
        meta,
        difficulty,
        tuning,
        capo,
        playKey,
        originalKey,
        beat,
        viewCount,
        favoriteCount,
      ];
}

class SongQuery extends Equatable {
  const SongQuery({
    this.genre,
    this.search,
    this.page,
    this.pageSize,
  });

  final String? genre;
  final String? search;
  final int? page;
  final int? pageSize;

  SongQuery copyWith({
    String? genre,
    String? search,
    int? page,
    int? pageSize,
  }) {
    return SongQuery(
      genre: genre ?? this.genre,
      search: search ?? this.search,
      page: page ?? this.page,
      pageSize: pageSize ?? this.pageSize,
    );
  }

  @override
  List<Object?> get props => [genre, search, page, pageSize];
}
